tools/rosbag/src/player.cpp

```diff
@@ -227,15 +236,20 @@ void Player::publish() {
             time_publisher_.setPublishFrequency(-1.0);
 
         paused_time_ = now_wt;
-
-        // Call do-publish for each message
+        
         foreach(MessageInstance m, view) {
             if (!node_handle_.ok())
                 break;
 
             doPublish(m);
+
+            if (skip_time_)
+                break;
```

Comment:
`skip_time_` ought to always be false entering into the `foreach`, and is only set false within `doPublish` if a skip key is pressed.  (Though there are one or more unwanted doPublishes whenever I use the skip function, I see a glitch in rqt_plot when playing a bag with a steadily increasing published value and skipping forward or backward)